ethcore/src/engines/authority_round/mod.rs

```diff
@@ -1060,12 +1098,14 @@ impl Engine<EthereumMachine> for AuthorityRound {
 			)));
 		}
 
-		// TODO [ToDr] Should this go from epoch manager?
-		// If yes then probably benign reporting needs to be moved further in the verification.
-		let set_number = header.number();
-
 		match verify_timestamp(&self.step.inner, header_step(header, self.empty_steps_transition)?) {
 			Err(BlockError::InvalidSeal) => {
+				let set_number = if self.immediate_transitions {
+					header.number()
+				} else {
+					self.epoch_set(header)?.1
```

Comment:
Is it guaranteed that parent block is already imported when we are doing `block_basic` verification? If not then zooming to the epoch set will most likely fail. Would be good to have a test case for this one.

Reply:
Didn't think of this, so it's possible that `verify_block_basic` is done in parallel and therefore the parent might not be imported yet? This verification must be done in phase 1 since we want `TemporarilyInvalid` errors to be emitted at this phase (otherwise it won't recover the sync process).